### PR TITLE
Replace improperly used asserts with warnings in prompt.lua

### PR DESCRIPTION
**Pull request by morethanoneanimal**
`assert('foo') `never fails because assert treats 'foo` like a condition, not as an error message. Replacing with warnings not to break existing configs.

Try:
`lua -e 'assert("Critical error!")'`

### CODE REVIEW

File: tests/examples/wibox/awidget/prompt/hooks.lua [hunk 1]
```diff
@@ -1,7 +1,8 @@
 local parent    = ... --DOC_NO_USAGE --DOC_HIDE
 local wibox     = require( "wibox"     ) --DOC_HIDE
 local awful     = { prompt = require("awful.prompt"),--DOC_HIDE
-                    util   = require("awful.util")}--DOC_HIDE
+                    util   = require("awful.util"),
```
Comment (Elv13) [inline tests/examples/wibox/awidget/prompt/hooks.lua]: the `--DOC_HIDE` is missing and this seems unrelated to the commit

**Comment by Elv13** [comment]
Otherwise 👍 

As discussed on IRC, the first version of that patch had `print()` instead of `assert()`. This was replaced at some point, but not correctly.

File: tests/examples/wibox/awidget/prompt/hooks.lua (hunk 1 above)
Comment (morethanoneanimal) [inline tests/examples/wibox/awidget/prompt/hooks.lua]: Fixed.

Comment (Elv13) [inline tests/examples/wibox/awidget/prompt/hooks.lua]: Not really, it's still unrelated to the commit. Spawn is never *used* in the test. While the example code has it, it's never called. Also, it should not be called in documentation test. We currently have no way to shim the commands.

Comment (morethanoneanimal) [inline tests/examples/wibox/awidget/prompt/hooks.lua]: It's not used but it needs to be declared, otherwise `awful.spawn` is nil.
Lua ignores nil values at the end of the table (`#{1,2,3,nil,nil,nil}` is 3). Without this declaration we end up passing hook with 2 arguments and printing warning.

File: lib/awful/prompt.lua [hunk 2]
```diff
@@ -518,10 +519,10 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                 hooks[key] = hooks[key] or {}
                 hooks[key][#hooks[key]+1] = v
             else
-                assert("The hook's 3rd parameter has to be a function.")
+                debug.print_warning("The hook's 3rd parameter has to be a function.")
             end
         else
-            assert("The hook has to have 3 parameters.")
+            debug.print_warning("The hook has to have 3 parameters.")
         end
```
Comment (blueyed) [inline lib/awful/prompt.lua]: Should be an error / return here IMHO?!

Comment (actionless) [inline lib/awful/prompt.lua]: it was stated in PR message what the previous behavior wanted to be preserved

Comment (blueyed) [inline lib/awful/prompt.lua]: I've read that, but would consider it to be a bug that it was not stopping here before.
Anyway, feel free to ignore it.

Comment (actionless) [inline lib/awful/prompt.lua]: mb then fix it properly but revert to this version (with warning) if few people will be complaining on a broken behavior?

Comment (morethanoneanimal) [inline lib/awful/prompt.lua]: I'm fine with asserts too, I wonder how much of an issue it would be, passing wrong parameters simply doesn't work or doesn't working as intended, so maybe the impact would be minimal?
I'd go with @actionless 's solution.

Comment (Elv13) [inline lib/awful/prompt.lua]: My opinion is still that code that didn't cause an error in 4.0 should not start to cause errors in 4.1. Beside, as I said on IRC, a warning is enough. It won't break anything if the table element count is wrong. Having an error is over-reacting.

**Comment by blueyed** [comment]
OK, thanks!
